integration_test/ops_agent_test/main_test.go

```diff
@@ -4822,29 +4972,57 @@ func TestRestartVM(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		cmdOut, err := gce.RunRemotely(ctx, logger, vm, getRecentServiceOutputForImage(vm.ImageSpec))
-		if err != nil {
-			t.Fatal(err)
-		}
+		isUAPPlugin := agents.IsOpsAgentUAPPlugin()
+		if isUAPPlugin {
```

Comment:
Would it make more sense to skip this test if we're in UAP world? Different from the systemd version, we don't actually expect the UAP plugin to be started automatically after a restart, so the behaviour this test checks for doesn't really apply. Maybe it's less confusing if we explicitly skip this test.

Reply:
Emm somehow I cant reply to your comment above. Let me reply here.
"In that case, is it possible to be more explicit here and call StartOpsAgentPlugin instead?"
- `StartOpsAgentPlugin` and `RestartOpsAgent` do different things for the agent plugin; `StartOpsAgentPlugin` starts the plugin grpc server itself. However, `RestartOpsAgent` sends a gRPC request to the plugin's `Start()` endpoint, to start the Ops Agent. 
In this case the plugin gRPC server is live already, I only need to start the Ops Agent. 





I am unsure what the RestartVM test is actually testing. If it's verifying that the agent installation always works even after a VM restart, then it makes sense to test for the plugin as well.